components/sup/src/manager/file_watcher.rs

```diff
@@ -2974,12 +2999,21 @@ mod tests {
             iterations
         }
 
-        fn spin_watcher(&self, setup: &mut WatcherSetup, iterations: u32) {
+        fn spin_watcher(&self, setup: &mut WatcherSetup, expected_event_count: u32) {
             let mut iteration = 0;
 
-            // After switching single_iteration() from recv() to try_recv(), this sleep is required
-            // for these tests to pass.
-            thread::sleep(Duration::from_secs(3));
+            let is_poll_watcher = match &setup.watcher.get_mut_underlying_watcher().real_watcher {
+                SupWatcher::Native(_watcher) => false,
+                SupWatcher::Fallback(_watcher) => true,
+            };
+
+            let mut iterations = expected_event_count;
+            if is_poll_watcher {
+                thread::sleep(Duration::from_secs(15));
+                iterations *= 5;
```

Comment:
can you explain why the iteration count needs to be bumped?

Reply:
"Through experimentation it was determined that the PollWatcher is less responsive
and emits more events than the NotifyWatcher.  The initial sleep used in NotifyWatcher
was not adequate to pass the tests and was increased as a result.  Also the number of iterations
required is larger for the PollWatcher case as there were intermediate events observed that would lead to
test case failure with the original iteration count used.  The test cases will fail if the desired events are not emitted, so the iteration count was increased to account for the increased number of events in the PollWatcher."